src/dynamodb.ts

```diff
@@ -0,0 +1,92 @@
+import { AccountRootPrincipal, Effect, PolicyDocument, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import {
+  AccessCapability,
+  canPrincipalsManageResources,
+  getAccessCapabilityFromValue,
+  IAccessSpec,
+  K9PolicyFactory,
+} from './k9policy';
+
+
+export interface K9DynamoDBResourcePolicyProps {
+  readonly k9DesiredAccess: Array<IAccessSpec>;
+}
+
+let SUPPORTED_CAPABILITIES = new Array<AccessCapability>(
+  AccessCapability.ADMINISTER_RESOURCE,
+  AccessCapability.READ_CONFIG,
+  AccessCapability.READ_DATA,
+  AccessCapability.WRITE_DATA,
+  AccessCapability.DELETE_DATA,
+);
+
+export const SID_DENY_EVERYONE_ELSE = 'DenyEveryoneElse';
+
+/**
+ * Generate a DynamoDB resource policy from the provided props that can be attached to DynamoDB
+ * resources, particularly tables & indices.
+ *
+ * @param props specifying desired access
+ * @return a PolicyDocument that can be attached to DynamoDB resources
+ */
+export function makeResourcePolicy(props: K9DynamoDBResourcePolicyProps): PolicyDocument {
```

Comment:
This works for our needs, but I was wondering if you still wanted to wrap this in a `grantAccessViaResourcePolicy` method to be consistent with s3.